### PR TITLE
Restructuring ASP.NET integration project

### PR DESCRIPTION
**Pull request by fabiocav**
Changing project layering;
- Removing ASP.NET dependencies from core projects (integration project only)
- Moving service registrations to integration project
- Changing project targets for integration
- 
### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: extensions/Worker.Extensions.Http.AspNetCore/src/ExtensionStartup.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Azure.Functions.Worker.Core.Http;
+using Microsoft.Azure.Functions.Worker.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+[assembly: WorkerExtensionStartup(typeof(ExtensionStartup))]
+
+namespace Microsoft.Azure.Functions.Extensions.Http.AspNetCore
+{
+    internal class ExtensionStartup : WorkerExtensionStartup
```
Comment (satvu) [inline extensions/Worker.Extensions.Http.AspNetCore/src/ExtensionStartup.cs:18]: I think this needs to be public for the generator - getting an error due to the protection level.

File: extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Azure.Functions.Worker.Core.Pipeline;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNet
+{
+    public static class HostBuilderExtensions
+    {
+        public static IHostBuilder ConfigureAspNetIntegration(this IHostBuilder builder)
+        {
+            var port = Environment.GetEnvironmentVariable("Azure_Functions_HttpProxyingPort");
```
Comment (brettsam) [inline extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs:19]: I was realizing that we kind of already have a precedent here of using `FUNCTIONS_` as the prefix for all of our special env vars -- suggest changing this (and making it all-caps).

File: extensions/Worker.Extensions.Http.AspNetCore/src/ExtensionStartup.cs (hunk 1 above)
Comment (fabiocav) [reply extensions/Worker.Extensions.Http.AspNetCore/src/ExtensionStartup.cs]: Yes. Opened #1404 to track enhancements to the experience, but taking a different approach with the changes.

File: extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs (hunk 2 above)
Comment (fabiocav) [reply extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs]: Yes, we should. I'm leaving some of this logic intact for now (restructuring things only) and will follow up with additional changes in different areas.

File: extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        // TODO: This should be modified to extend IFunctionsWorkerApplicationBuilder.
+        /// <summary>
+        /// Configures the worker to use the ASP.NET Core integration, enabling advanced HTTP features.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHostBuilder"/> to configure.</param>
+        /// <returns>The same instance of the <see cref="IHostBuilder"/> for chaining.</returns>
+        public static IHostBuilder ConfigureAspNetCoreIntegration(this IHostBuilder builder)
+        {
+            // TODO: Update this logic. Port should always come through configuration.
```
Comment (fabiocav) [inline extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs:24]: This is temporary. The client registration currently requires the following:

``` csharp

var host = new HostBuilder()
    .ConfigureFunctionsWorkerDefaults(builder =>
    {
        builder.UseAspNetCoreIntegration();
    })
    .ConfigureAspNetCoreIntegration()
    .Build();
```

**Review by satvu — APPROVED** [review]

**Pull request merged**